### ISSUE TITLE
Enable hardware acceleration when decoding H264 stream

### ISSUE THREAD

**Issue by jasaw** [Motion-Project/motion#560]
1. version [x.y.z, hash, other]: git-ccbe2f59272b029b49a106224e8bde7114da0814
2. installed as a package or compiled from sources [deb, rpm, git, other]: all
3. standalone or part of third party [motion, MotionEyeOS, other]: all
4. video stream source [V4L (card or USB), net cam (mjpeg, rtsp, other), mmal]: H264 net cam
5. hardware [x86, ARM, other]: all
6. operating system [Linux (which), FreeBSD, other]: all

This is a feature request to enable hardware acceleration when decoding H264 stream. We could use ffmpeg's `ff_find_hwaccel` function to find a suitable hardware accelerated decoder and use it. If hardware accelerated decoder is not found, no harm done. To make the configuration more flexible, we could introduce a new config item to enable/disable hardware acceleration for a particular netcam.

This would benefit users who run motion on a Raspberry Pi, like motionEye(OS) users.

One way to implement this is to add something like this to netcam_rtsp.c file.
`rtsp_data->codec_context->hwaccel = ff_find_hwaccel(avctx->codec->id, avctx->pix_fmt);`

**Comment by Mr-Dave** [comment]
This sounds like an extremely reasonable addition.  Can you confirm it works?  Have you experienced any freezes or other downsides such as not working under various versions of ffmpeg, etc?

**Comment by jasaw** [comment]
Sorry about the slow response, I've been swamped with work recently. I have **not** tested the `ff_find_hwaccel` function, but I have tested it with `h264_mmal` hardware accelerated decoder and it seems to work well (haven't got time to test extensively).

To choose the `h264_mmal` decoder, I just moved `h264_mmal` up the priority list in ffmpeg so it gets selected as the default decoder. I did this as a quick hack to prove the concept, prove that it works.

**Comment by Mr-Dave** [comment]
I just tried the snippet of code provided and it appears to require the internal ffmpeg function `ff_find_hwaccel` that is not part of the public api.   

Looking quickly at the ffmpeg docs, I did not see any samples invoking this and the limited documentation that was there leads me to believe this would not be a simple change. 

I am going to close this issue for now and if you have a sample that works against the standard ffmpeg libs and embeds it in Motion, submit a PR and we can test/evaluate/merge it.
